plat/common/include/x86/paging.h

```diff
@@ -266,6 +269,18 @@ pgarch_init(void)
 	}
 #endif /* PT_LEVELS == 5 */
 
+	/* Check for PAT support */
+	ukarch_x86_cpuid(0x1, 0, &eax, &ebx, &ecx, &edx);
+	if (unlikely(!(edx & X86_CPUID1_EDX_PAT))) {
+		uk_pr_crit("Page table attributes are not supported.\n");
+		return -ENOTSUP;
+	}
+	/* Write default value described in Intel's Software Developer
+	 * Manual to the MSR, in case BIOS modified them.
+	 * (See Vol 3, 11.12.4 Programming the PAT)
+	 */
+	wrmsrl(0x277, 0x0007040600070406);
```

Comment:
I agree with Alex here that this is still a bit cumbersome to read. Please cherry-pick the commits d7333a30816d7e419c6af13e716706e784c9f606 and a0695d6e00490876baef88a97ba6e2d2ab4b6d23 from this branch https://github.com/marcrittinghaus/unikraft/tree/pr_788_writecombine.

Then you can write:
```C
/* Reset PAT to default value */
wrmsrl(X86_MSR_PAT, X86_PAT_DEFAULT);
```